src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -487,10 +492,15 @@ BlobMetadata buildAndUpload(String filePath, List<List<ChannelData<T>>> blobData
 
     // Construct the blob along with the metadata of the blob
     BlobBuilder.Blob blob = BlobBuilder.constructBlobAndMetadata(filePath, blobData, bdecVersion);
-    if (buildContext != null) {
-      buildContext.stop();
-    }
-    return upload(filePath, blob.blobBytes, blob.chunksMetadataList);
+
+    return buildContext != null
+        ? upload(
+            filePath,
+            blob.blobBytes,
+            blob.chunksMetadataList,
+            TimeUnit.NANOSECONDS.toMillis(buildContext.stop()))
+        : upload(
+            filePath, blob.blobBytes, blob.chunksMetadataList, BlobMetadata.DEFAULT_BLOB_LATENCY);
```

Comment:
nit: get the time first and then call upload

buildlatency = !=null? a: b
return upload (xx,xx,xx,a)